monty/shutil.py

```diff
@@ -75,15 +77,27 @@ def compress_file(
         filepath (str | Path): Path to file.
         compression (str): A compression mode. Valid options are "gz" or
             "bz2". Defaults to "gz".
+        target_dir (str | Path): An optional target dir where the result compressed
+            file would be stored. Defaults to in-place compression.
     """
     filepath = Path(filepath)
-    if compression not in ["gz", "bz2"]:
+    target_dir = str(target_dir)
```

Comment:
Might need a more elegant way here. If using `Path` for `target_dir` directly, the default `if Path("")` would becomes `True`